front/pages/w/[wId]/builder/agents/new.tsx

```diff
@@ -0,0 +1,181 @@
+import type { InferGetServerSidePropsType } from "next";
+import type { ParsedUrlQuery } from "querystring";
+
+import AgentBuilder from "@app/components/agent_builder/AgentBuilder";
+import { AgentBuilderProvider } from "@app/components/agent_builder/AgentBuilderContext";
+import {
+  buildInitialActions,
+  getAccessibleSourcesAndApps,
+} from "@app/components/assistant_builder/server_side_props_helpers";
+import type {
+  AssistantBuilderInitialState,
+  BuilderFlow,
+} from "@app/components/assistant_builder/types";
+import { BUILDER_FLOWS } from "@app/components/assistant_builder/types";
+import AppRootLayout from "@app/components/sparkle/AppRootLayout";
+import { throwIfInvalidAgentConfiguration } from "@app/lib/actions/types/guards";
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import { generateMockAgentConfigurationFromTemplate } from "@app/lib/api/assistant/templates";
+import config from "@app/lib/api/config";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
+import { getFeatureFlags } from "@app/lib/auth";
+import { withDefaultUserAuthRequirements } from "@app/lib/iam/session";
+import { useAssistantTemplate } from "@app/lib/swr/assistants";
+import type {
+  AgentConfigurationType,
+  AppType,
+  DataSourceViewType,
+  PlanType,
+  SpaceType,
+  SubscriptionType,
+  TemplateAgentConfigurationType,
+  WorkspaceType,
+} from "@app/types";
+
+function getDuplicateAndTemplateIdFromQuery(query: ParsedUrlQuery) {
+  const { duplicate, templateId } = query;
+
+  return {
+    duplicate: duplicate && typeof duplicate === "string" ? duplicate : null,
+    templateId:
+      templateId && typeof templateId === "string" ? templateId : null,
+  };
+}
+
+export const getServerSideProps = withDefaultUserAuthRequirements<{
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  plan: PlanType;
+  spaces: SpaceType[];
+  dataSourceViews: DataSourceViewType[];
+  dustApps: AppType[];
+  mcpServerViews: MCPServerViewType[];
+  actions: AssistantBuilderInitialState["actions"];
+  agentConfiguration:
+    | AgentConfigurationType
+    | TemplateAgentConfigurationType
+    | null;
+  flow: BuilderFlow;
+  baseUrl: string;
+  templateId: string | null;
+}>(async (context, auth) => {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const subscription = auth.subscription();
+  if (!owner || !plan || !auth.isUser() || !subscription) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const featureFlags = await getFeatureFlags(owner);
+  if (!featureFlags.includes("agent_builder_v2") || !auth.isBuilder()) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const { spaces, dataSourceViews, dustApps, mcpServerViews } =
```

Comment:
This is basically duplicated from the builder/assistants/new.tsx file and everything might not be needed